docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,55 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run. 
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed. 
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task. 
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.  
+
+== Solution
+
+`ods-start` provides a file called `$WORKING_DIR/.ods/git-dir-commit-sha` which contains the git commit sha of the working directory (no whitespace or newlines).
+
+Build tasks check first whether their artifacts are available under the working dir commit sha. 
+If so the output is made available in the `output-dir` and the task sets as a result that it 'reused' the prior build.
+This can happen outside the individual build scripts which in this case would no longer be invoked and thus do not need to code for the scenario.
+
+A new script or go program would be provided instead that implements the mentioned logic and would be available in all tasks images.
```

Comment:
Would this logic interface with Nexus or do you envision a local logic. I would be hesitant about any Nexus interaction. ODS pipeline is designed specifically to avoid this interface in tasks between start and finish.

Reply:
If we want to avoid nexus interactions except in start or finish, then this would have to happen in `ods-start`. So far I assumed that this might be task specific, but I am not sure.